src/common/utils/storage.ts

```diff
@@ -0,0 +1,79 @@
+import { useState } from 'react';
+
+// a single place to define all used localStorage keys, to avoid collisions or hardcoded strings
+export enum StorageKeys {
+  COUNTY_MAP_TYPE = 'COUNTY_MAP_TYPE',
+  FULLSTORY_RECORD = 'FULLSTORY_RECORD',
+}
+
+// From https://developer.mozilla.org/en-US/docs/Web/API/Web_Storage_API/Using_the_Web_Storage_API#testing_for_availability
+export function storageAvailable(storageType: string) {
+  let storage;
+  const testKey = '__storage_test__';
+  try {
+    // @ts-ignore
+    storage = window[storageType];
+    storage.setItem(testKey, testKey);
+    storage.removeItem(testKey);
+    return true;
+  } catch (e) {
+    return false;
+  }
+}
+
+// Hook from https://usehooks.com/useLocalStorage/
+// but updated to fallback to setState if localstorage is not available.
+export function useLocalStorage<T>(
```

Comment:
I was thinking I could update this to have the fallback built-in (i.e., just don't try to use localstorage if it isn't available).  Would that  be worth doing?